apps/explorer/src/pages/transaction-result/programmable-transaction-view/Command.tsx

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+    type MoveCallSuiCommand,
+    type SuiArgument,
+    type SuiMovePackage,
+} from '@mysten/sui.js';
+import { type ReactNode } from 'react';
+
+import { flattenSuiArguments } from './utils';
+
+import { ObjectLink } from '~/ui/InternalLink';
+
+export interface ICommandProps<T> {
+    type: string;
+    data: T;
+}
+
+function CommandContent({
+    type,
+    children,
+}: {
+    type: string;
+    children?: ReactNode;
+}) {
+    return (
+        <>
+            <div
+                data-testid="programmable-transactions-command-label"
+                className="text-heading6 font-semibold text-steel-darker"
+            >
+                {type}
+            </div>
+            {children && (
+                <div
+                    data-testid="programmable-transactions-command-content"
+                    className="text-bodyMedium pt-2 font-medium text-steel-dark"
+                >
+                    {children}
+                </div>
+            )}
+        </>
+    );
+}
+
+function ArrayArgument({
+    type,
+    data,
+}: {
```

Comment:
I'm fine with this being outlined explicitly, but seems like this should use `ICommandProps` if that still exists (or remove that type helper.